app/assets/javascripts/discourse/routes/user.js.es6

```diff
@@ -39,7 +41,11 @@ export default Discourse.Route.extend({
       return this.currentUser;
     }
 
-    return Discourse.User.create({ username: params.username });
+    if (params.username.match(USERNAME_ROUTE_REGEXP)) {
```

Comment:
Does `/u/%E7%8B%AE%E5%AD%90/summary` (for the user `狮子`) still work? This will probably break Unicode usernames unless `params.username` contains the URL encoded username.